doc/user/content/overview/architecture.md

```diff
@@ -8,23 +8,124 @@ menu:
 
-The architecture is broken into three logical components:
+[todo]: I feel like I said stuff about scalability but not really about how virtual time ensures e.g. strict serializability.
+
+## Regions
+
+All of the components of the Materialize platform are scoped to a single AWS region and cannot be shared between regions. [TODO: is this true?] This is important because we'll soon talk about sharing resources and its important to remember that we cannot share anything across regions.
+
+## Layers
+
+Now we're ready to dive deeper into each of the three layers. Each layer has its own SQL nouns and concepts. This section will overview each layer, what it does, and how all of the nouns and concepts work.
 
 ### Storage
 
+Storage is responsible for ensuring that data read from external systems (e.g. a Kafka topic), are granted a timestamp and then durably persisted to storage (currently S3). Storage is also responsible for writing data from Materialize back to external systems (e.g. Kafka again). Storage handles connecting to/authenticating these external systems and converting data from/to external formats (e.g. Protobuf) to/from the Materialize internal binary representation of data.
+
+The key concepts and nouns in storage that users have to interact with are `CONNECTION`s, `SOURCE`s, and `SINK`s.
+
+#### Connections
+
+A connection describes how to connect and authenticate to an external system you want Materialize to read data from/to. A connection is a logical entity in Materialize in that it exists purely to make subsequent usage of Materialize easier for the user, and creating a connection itself doesn't cause anything to happen.
+
+#### Sources
+
+Sources describe external systems you want Materialize to read data from, and provide details about how to decode and interpret that data. Importantly, and unlike connections, as soon as you create a source, Materialize starts reading from the specified external system, and decoding, interpreting and ultimately writing the data out to persistent storage. The amount of computational resources (CPU and memory) available to handle all of the data from a source can be scaled up and down by changing the source's size with `ALTER SOURCE`.
+
+#### Sinks
+
+Sinks are the inverse of sources and represent a connection to an external stream where Materialize outputs data from its persistent storage. Sinks don't read directly from the compute layer [todo: is this true?]. Sinks, like sources begin their works immediately. When a user defines a sink over a materialized view, source, or table, Materialize automatically generates the required schema and writes down the stream of changes to that materialized view, source, or table. The amount of computational resources (CPU and memory) available to encode all of the data that needs to be written can be scaled up and down by changing the sink's size with `ALTER SINK`.
```

Comment:
> Sinks don't read directly from the compute layer [todo: is this true?].

Sinks can only be created on top of persisted objects, so AFAIU these read out of storage collections. I don't think it's important to call this out if you mention that sinks can be defined over sources, materialized views and tables.